internal/plugin/convert/diagnostics.go

```diff
@@ -93,13 +93,13 @@ func AttributePathToPath(ap *tftypes.AttributePath) cty.Path {
 		return p
 	}
 	for _, step := range ap.Steps() {
-		switch step.(type) {
+		switch step := step.(type) {
 		case tftypes.AttributeName:
-			p = p.GetAttr(string(step.(tftypes.AttributeName)))
+			p = p.GetAttr(string(step))
 		case tftypes.ElementKeyString:
-			p = p.Index(cty.StringVal(string(step.(tftypes.ElementKeyString))))
+			p = p.Index(cty.StringVal(string(step)))
 		case tftypes.ElementKeyInt:
-			p = p.Index(cty.NumberIntVal(int64(step.(tftypes.ElementKeyInt))))
```

Comment:
So, to understand: here we are getting rid of a "double casting". Before it was going (for example) from `interface{}` to `tftypes.AttributeName` to `string`. Now, just `interface{}` to `string`. Correct?

Reply:
Correction: it's not `casting`. You are removing the _type assertion_ because the assertion was already done by the `switch step.(type)`. So, it's safe to then do just the _type conversion_.

Reply:
Here's the original issue reported in the linter:

```
internal/plugin/convert/diagnostics.go:96:10: S1034: assigning the result of this type assertion to a variable (switch step := step.(type)) could eliminate type assertions in switch cases (gosimple)
                switch step.(type) {
                       ^
internal/plugin/convert/diagnostics.go:98:25: S1034(related information): could eliminate this type assertion (gosimple)
                        p = p.GetAttr(string(step.(tftypes.AttributeName)))
                                             ^
internal/plugin/convert/diagnostics.go:100:37: S1034(related information): could eliminate this type assertion (gosimple)
                        p = p.Index(cty.StringVal(string(step.(tftypes.ElementKeyString))))
                                                         ^
internal/plugin/convert/diagnostics.go:102:39: S1034(related information): could eliminate this type assertion (gosimple)
                        p = p.Index(cty.NumberIntVal(int64(step.(tftypes.ElementKeyInt))))
                                                           ^
```

There is a fun feature in the Go specification that allows `switch` statements to automatically handle type assertions when using the `.(type)` syntax with assignment. It's really handy for simplifying code that needs to handle multiple types, if you haven't run across it before to avoid the duplicate type handling within a case.

Reply:
yep yep: essentially, by the time we are in any of the specific cases of the switch, the assertion has already happened and we can avoid doing it again. Makes sense.